addons/microsoft_calendar/models/res_users.py

```diff
@@ -20,6 +20,8 @@ class User(models.Model):
 
     microsoft_calendar_sync_token = fields.Char('Microsoft Next Sync Token', copy=False)
     microsoft_synchronization_stopped = fields.Boolean('Outlook Synchronization stopped', copy=False)
+    # Save the last sync date with Microsoft to sync only events created/updated after the last sync.
+    last_sync_date = fields.Datetime('Last Sync Date', copy=False, store=True)
```

Comment:
Warning ! some database have a lot of user (all portal user are here and we have thousands in our Odoo.com database. Adding a field like that that will be updated a lot is a bad idea. Updating the row will lock it and it will badly affect the database :'(.